src/cargo/util/toml.rs

```diff
@@ -680,6 +681,12 @@ fn process_dependencies<F>(cx: &mut Context,
             }
             TomlDependency::Detailed(ref details) => details.clone(),
         };
+
+        if details.version.is_none() && details.path.is_none() && details.git.is_none() {
+            warnings.push(format!("warning: Dependency ({}) specified without providing a local \
```

Comment:
Question for reviewer: Is there a convention for phrasing to indicate this really *should* be changed and may become a hard error in the future?

Similarly, is there anything (comment, annotation, etc) that should be left here in the code as a marker for future cleanup?